object-storage-core/src/main/java/io/micronaut/objectstorage/UploadRequest.java

```diff
@@ -49,87 +54,107 @@ static UploadRequest fromBytes(@NonNull byte[] bytes,
      */
+    @NonNull
     Optional<Long> getContentSize();
 
     /**
      * @return an input stream of the object to be stored.
      */
+    @NonNull
     InputStream getInputStream();
 
     /**
      * Upload request implementation using {@link java.io.File}.
      */
     class FileUploadRequest implements UploadRequest {
 
+        private static final Logger LOG = LoggerFactory.getLogger(FileUploadRequest.class);
+
         private final String keyName;
         private final String contentType;
         private final Path path;
 
-        public FileUploadRequest(Path localFilePath) {
+        public FileUploadRequest(@NonNull Path localFilePath) {
             this(localFilePath, localFilePath.getFileName().toString(), null,
                 URLConnection.guessContentTypeFromName(localFilePath.toFile().getName()));
         }
 
-        public FileUploadRequest(Path localFilePath, String prefix) {
+        public FileUploadRequest(@NonNull Path localFilePath,
+                                 @Nullable String prefix) {
             this(localFilePath, localFilePath.toFile().getName(), prefix,
                 URLConnection.guessContentTypeFromName(localFilePath.toFile().getName()));
         }
 
-        public FileUploadRequest(Path localFilePath,
-                                 String keyName,
+        public FileUploadRequest(@NonNull Path localFilePath,
+                                 @NonNull String keyName,
                                  @Nullable String prefix,
                                  @Nullable String contentType) {
             this.keyName = prefix != null ? prefix + "/" + keyName : keyName;
             this.contentType = contentType;
             this.path = localFilePath;
         }
 
+        @NonNull
         public File getFile() {
             return path.toFile();
         }
 
+        @NonNull
         public Path getPath() {
             return path;
         }
 
+        @NonNull
         public String getAbsolutePath() {
             return path.toAbsolutePath().toString();
         }
 
         @Override
+        @NonNull
         public Optional<String> getContentType() {
             return Optional.ofNullable(contentType);
         }
 
         @Override
+        @NonNull
         public String getKey() {
             return keyName;
         }
 
         @Override
+        @NonNull
         public Optional<Long> getContentSize() {
             try {
                 return Optional.of(Files.size(path));
             } catch (IOException e) {
+                if (LOG.isErrorEnabled()) {
+                    LOG.error("Error calculating the size of the file.", e);
+                }
                 return Optional.empty();
             }
         }
 
         @Override
+        @NonNull
         public InputStream getInputStream() {
             try {
                 return Files.newInputStream(path);
             } catch (IOException e) {
+                if (LOG.isErrorEnabled()) {
```

Comment:
shouldn't log here since you're throwing the exception which will also get logged. delete or change to `throw new ObjectStorageException("Error creating an input stream to read from the file", e);`

Reply:
[I've removed the logging statement. ](https://github.com/micronaut-projects/micronaut-object-storage/pull/85#issuecomment-1231885433)